exp_test.go

```diff
@@ -0,0 +1,228 @@
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber string `index:"2,keepzero"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "01104000000000000000000000000000000000",
+			},
+
+			// Tests for *string type
+			{
+				name: "struct with *string type and value set",
+				input: struct {
+					MTI                  string  `index:"0"`
+					PrimaryAccountNumber *string `index:"2"`
+				}{
+					MTI:                  "0110",
+					PrimaryAccountNumber: &panStr,
+				},
+				expectedPackedString: "011040000000000000000000000000000000164242424242424242",
+			},
+			{
+				name: "struct with *string type and no value",
+				input: struct {
+					MTI                  string  `index:"0"`
+					PrimaryAccountNumber *string `index:"2"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "011000000000000000000000000000000000",
+			},
+			{
+				name: "struct with *string type, no value and keepzero tag",
+				input: struct {
+					MTI                  string  `index:"0"`
+					PrimaryAccountNumber *string `index:"2,keepzero"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "011000000000000000000000000000000000",
+			},
+
+			// Tests for int type
+			{
+				name: "struct with int type and value set",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber int    `index:"2"`
+				}{
+					MTI:                  "0110",
+					PrimaryAccountNumber: panInt,
+				},
+				expectedPackedString: "011040000000000000000000000000000000164242424242424242",
+			},
+			{
+				name: "struct with int type and no value",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber int    `index:"2"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "011000000000000000000000000000000000",
+			},
+			{
+				name: "struct with int type, no value and keepzero tag",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber int    `index:"2,keepzero"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "011040000000000000000000000000000000010",
+			},
+
+			// Tests for *int type
+			{
+				name: "struct with *int type and value set",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber *int   `index:"2"`
+				}{
+					MTI:                  "0110",
+					PrimaryAccountNumber: &panInt,
+				},
+				expectedPackedString: "011040000000000000000000000000000000164242424242424242",
+			},
+			{
+				name: "struct with *int type and no value",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber *int   `index:"2"`
+				}{
+					MTI: "0110",
+				},
+				expectedPackedString: "011000000000000000000000000000000000",
+			},
+			{
+				name: "struct with *int type, no value and keepzero tag",
+				input: struct {
+					MTI                  string `index:"0"`
+					PrimaryAccountNumber *int   `index:"2,keepzero"`
```

Comment:
same here. I think we should create a new **int** and set the field value to `0`.